reconfiguration bug / test failure

If the sleep linked to below is removed, the test fails, because the validators never advance to the next epoch.

https://github.com/MystenLabs/sui/blob/5ad8e2a092e33a49e137db4a02067bbf40653e22/crates/sui/tests/reconfiguration_tests.rs#L108

To reproduce, comment out the sleep and run

`cargo simtest basic_reconfig_end_to_end_test`

I added more logging to consensus adapter in #7125.

This surfaces the issue with narwhal on one of the nodes that can't send EndOfPublish message - the narwhal seem not to listen to internal GRPC for submitting transaction, the call to this method simply hangs forever:

```
2023-01-04T19:39:21.318396Z ERROR sui_core::consensus_adapter: crates/sui-core/src/consensus_adapter.rs:359: Error submitting transaction to own narwhal worker: ConsensusConnectionBroken("Timeout when sending to narwhal")
2023-01-04T19:39:24.945378Z  WARN sui_core::consensus_adapter: crates/sui-core/src/consensus_adapter.rs:343: Still waiting 120 seconds for transaction EndOfPublish(k#b3fd5efb5c872403a4ad13f1199513606d23354a845249aea157564c23ff3c15c6c9e652b72d740e8cacb93cbcc0686b0ec0cc0c1c760bfb42b436519fc5a8054b76533a97a9f23987d097ef96b1f2526bb2073fa10a7719941772347ef274f4) to commit in narwhal
2023-01-04T19:39:28.462712Z ERROR sui_core::consensus_adapter: crates/sui-core/src/consensus_adapter.rs:359: Error submitting transaction to own narwhal worker: ConsensusConnectionBroken("Timeout when sending to narwhal")
2023-01-04T19:39:35.575993Z ERROR sui_core::consensus_adapter: crates/sui-core/src/consensus_adapter.rs:359: Error submitting transaction to own narwhal worker: ConsensusConnectionBroken("Timeout when sending to narwhal")

```

I think someone on narwhal side should take a look and see if this is something that can happen in production
cc @laura-makdah @akichidis

--- a/crates/sui/tests/reconfiguration_tests.rs
+++ b/crates/sui/tests/reconfiguration_tests.rs
@@ -109,10 +109,6 @@ async fn advance_epoch_tx_test_impl(
 #[sim_test]
 async fn basic_reconfig_end_to_end_test() {
     let authorities = spawn_test_authorities([].into_iter(), &test_authority_configs()).await;
-
-    // TODO: If this line is removed, the validators never advance to the next epoch
-    tokio::time::sleep(Duration::from_secs(1)).await;
-
     trigger_reconfiguration(&authorities).await;
 }
 

